sdk/storage/azure-storage-file-share/azure/storage/fileshare/_serialize.py

```diff
@@ -46,25 +47,58 @@ def get_source_conditions(kwargs):
         source_if_none_match=if_none_match or kwargs.pop('source_if_none_match', None)
     )
 
-
-def validate_copy_mode(copy_mode, permission, permission_key):
-    if copy_mode == "source":
-        if permission or permission_key:
-            raise ValueError("Copy mode should be 'override' if permission/permission_key is specified")
-    elif copy_mode == "override":
-        if not (permission or permission_key):
-            raise ValueError("permission/permission_key should be specified with 'override' mode")
-    elif not copy_mode:
-        if permission_key or permission_key:
-            raise ValueError("permission/permission_key shouldn't be specified without copy mode")
-    else:
-        raise ValueError("Invalid copy mode")
-
-
 def get_access_conditions(lease):
     # type: (Optional[Union[ShareLeaseClient, str]]) -> Union[LeaseAccessConditions, None]
     try:
         lease_id = lease.id # type: ignore
     except AttributeError:
         lease_id = lease # type: ignore
     return LeaseAccessConditions(lease_id=lease_id) if lease_id else None
+
+
+def get_smb_properties(kwargs):
+    # type: (Optional[Union[Dict, FileSmbProperties]]) -> Dict[str, Any]
+    ignore_read_only = kwargs.pop('ignore_read_only', None)
+    set_archive_attribute = kwargs.pop('set_archive_attribute', None)
+    file_smb_properties = kwargs.pop('file_smb_properties', None)
+    if not any([ignore_read_only, set_archive_attribute, file_smb_properties]):
+        return {}
+    try:
+        smb_props = file_smb_properties.__dict__  # FileSmbProperties object
+    except AttributeError:
+        smb_props = file_smb_properties  # Dictionary
+
+    file_permission = smb_props.get('file_permission')
+    file_permission_key = smb_props.get('file_permission_key')
+    file_attributes = smb_props.get('file_attributes')
+    file_creation_time = smb_props.get('file_creation_time') or ""
+    file_last_write_time = smb_props.get('file_last_write_time') or ""
+    file_permission_copy_mode = None
+    file_permission = _get_file_permission(file_permission, file_permission_key, "inherit")
```

Comment:
if we set "inherit" as default value, that means file and file_permission_key will not be both empty, while it is allowed to leave them both empty, then the permission will be inherited from the target's parent

Reply:
Ah - thanks - I was just taking the default value from the generated code.
Should I set default as None?

Reply:
Yes, I think we need to set it to None

Reply:
Fixed.